food-truck-frontend/src/components/Form.tsx

```diff
@@ -35,48 +36,67 @@ class Form extends Component<Props, State> {
     super(props);
 
     this.state = {formData: {}};
-    React.Children.forEach(this.props.children, c => {
-      // Ensure a null didn't slip through (can happen in dynamic forms)
-      if (c === null)
+    recursiveMap(this.props.children, node => {
+      if (!React.isValidElement(node))
         return;
-
-      // Parse child
-      const child = c as ReactElement;
-      if (child.props.name != null) {
-        if (child.props.value != null) {
-          this.state.formData[child.props.name] = child.props.value;
-        } else if (child.props.defaultValue != null) {
-          this.state.formData[child.props.name] = child.props.defaultValue;
-        } else {
-          this.state.formData[child.props.name] = "";
-        }
+      const e = node as ReactElement;
+
+      if (e.props.name) {
+        this.state.formData[e.props.name] =
+            e.props.value ??
+            e.props.defaultValue ??
+            e.props.checked ??
+            "";
       }
     });
 
+    //console.log("Initial state: ");
```

Comment:
If we have time, we should probably delete commented-out code. But I'm not sure we have time.